### PR TITLE
chore(jotai): migrate to jotai v2

### PR DESCRIPTION
**Pull request by gael-boyenval**
## Description

This migrates Jotai to the v2 release. It changes the way that atoms are shared between Centreon and Modules by using a global (which contains atoms defined in components subtree). The store is created in Centreon Web then shared to Modules by passing it down to props.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/packages/ui-context/src/defaults.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { User, ThemeMode, ListingVariant } from './types';
 
 const defaultUser: User = {
-  alias: '',
+  alias: 'toto',
```
Comment (kduret) [inline centreon/packages/ui-context/src/defaults.ts:4]: since it is not a test file, is it expected ?

Comment (Thebarda) [reply centreon/packages/ui-context/src/defaults.ts:4]: Yeah I'll remove it

File: centreon/package.json [hunk 2]
```diff
@@ -12,8 +12,8 @@
     "test": "jest",
     "test:watch": "pnpm test --watch",
     "test:coverage": "jest --coverage",
-    "cypress:cli": "./cypress/scripts/cypress-cli.sh",
-    "cypress:cli:updateSnapshot": "./cypress/scripts/cypress-cli.sh --env updateSnapshots=true",
+    "cypress:cli": "cypress run --component --browser=chrome",
+    "cypress:cli:updateSnapshot": "pnpm cypress:cli --env updateSnapshots=true",
```
Comment (Thebarda) [inline centreon/package.json:16]:
```suggestion
    "cypress:cli": "cypress run --component --    "cypress:cli": "./cypress/scripts/cypress-cli.sh",
    "cypress:cli:updateSnapshot": "./cypress/scripts/cypress-cli.sh --env updateSnapshots=true",
```

File: centreon/packages/ui-context/src/defaults.ts (hunk 1 above)
Comment (Thebarda) [inline centreon/packages/ui-context/src/defaults.ts:4]:
```suggestion
  alias: '',
```

**Review by Thebarda — APPROVED** [review]

File: centreon/www/front_src/src/Main/index.test.tsx [hunk 3]
```diff
@@ -294,7 +294,8 @@ describe('Main', () => {
     window.history.pushState({}, '', '/');
   });
 
-  it('displays the login page when the path is "/login" and the user is not connected', async () => {
+  // To migrate to Cypress
+  it.only('displays the login page when the path is "/login" and the user is not connected', async () => {
```
Comment (kduret) [inline centreon/www/front_src/src/Main/index.test.tsx:298]: why `only` is used ?
other tests are not compatible with jotai v2 and must be migrated to cypress ?

**Review by SoldierBoy31 — APPROVED** [review]

File: centreon/www/front_src/src/Main/index.test.tsx (hunk 3 above)
Comment (Thebarda) [reply centreon/www/front_src/src/Main/index.test.tsx:298]: Yes they have to be migrate to Cypress, I have created a ticket for this specific thing.

**Review by kduret — APPROVED** [review]

**Pull request merged**